backend/queues/listingCreatedProcessor.js

```diff
@@ -0,0 +1,181 @@
+ * @throws
+ */
+async function processor(job) {
+  const queueLog = (progress, str) => {
+    job.log(str)
+    job.progress(progress)
+  }
+  const jobId = `${job.queue.name}-${job.id}` // Prefix with queue name since job ids are not unique across queues.
+
+  const { txHash, fromAddress, shopId } = job.data
+  log.info(
+    `Waiting for ListingCreated tx. ShopId: {shopId} Hash: ${txHash} fromAddress: ${fromAddress} `
+  )
+  let confirmation
+
+  try {
+    // Load the associated shop.
+    const shop = await Shop.findOne({ where: { id: shopId } })
+    if (!shop) {
+      throw new Error(`Failed loading shop with id ${shopId}`)
+    }
+
+    const network = await Network.findOne({
+      where: { networkId: shop.networkId, active: true }
+    })
+    const provider = new ethers.providers.JsonRpcProvider(network.provider)
+    const marketplace = new ethers.Contract(
+      network.marketplaceContract,
+      marketplaceAbi,
+      provider
+    )
+
+    // Load the transaction from the DB.
+    const transaction = await Transaction.findOne({
+      networkId: network.id,
+      shopId,
+      fromAddress,
+      hash: txHash
+    })
+    if (!transaction) {
+      throw new Error(
+        `No transaction found in the DB with hash ${txHash} for shop ${shopId} and wallet ${fromAddress}`
+      )
+    }
+
+    // Load the tx from the blockchain based on its hash.
+    queueLog(25, 'Loading tx from the blockchain')
+    const tx = await provider.getTransaction(txHash)
+    if (!tx) {
+      throw new Error(`Transaction with hash ${txHash} not found`)
+    }
+    log.info(`Loaded tx with hash ${txHash} from the network`)
+
+    // Wait for the tx to get mined.
+    queueLog(50, `Waiting for tx ${txHash} to get confirmed`)
+    log.info('Waiting for tx ${txHash} confirmation...')
+    confirmation = await _waitForListingCreatedTxConfirmation(marketplace, tx)
+    const { receipt, listingId } = confirmation
+
+    if (receipt.status) {
+      // Update the transaction in the DB.
+      const lid = new ListingID(listingId, network.networkId).toString() // fully qualified listingId.
+      await transaction.update({
+        status: TransactionStatuses.Confirmed,
+        blockNumber: receipt.blockNumber,
+        listingId: lid,
+        jobId
+      })
+      // Check the shop still does not have a listingId (a concurrent ListingCreated tx could
+      // have been mined while we were waiting for this tx to get mined).
+      await shop.load()
+      if (shop.listingId) {
+        log.info(
+          `ListingId ${shop.listingId} linked to shop ${shopId} while waiting for ${txHash}.`
+        )
+      } else {
+        await shop.update({ listingId: lid })
+        log.info(
+          `Shop ${shopId}: tx ${txHash} mined, updated listingId to ${lid}`
+        )
+      }
+    } else {
+      log.info(`Shop ${shopId}: tx with hash ${txHash} was reverted.`)
+      await transaction.update({
```

Comment:
Wonder if we should send to Sentry here so we're aware if this happens

Reply:
Good idea. Will do.